src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapDecoder.cs

```diff
@@ -1253,21 +1234,6 @@ out decoder
             string decoderMimeTypes;
             clsId = GetCLSIDFromDecoder(decoderHandle, out decoderMimeTypes);
 
-            // If the mime type of the file does not match the associated decoder,
-            // and if we are in a Partial trust scenario, throw!
```

Comment:
what happens now that we don't throw?

Reply:
Logic now just always follows the full trust path (we only threw when the demand failed).